libs/sim/simulator.rs

```diff
@@ -69,43 +94,45 @@ impl Simulator {
         self.add_node(id, node);
     }
 
-    pub fn add_edge(&mut self, delay: u32, element1: &str, element2: &str, unidirectional: bool) {
-        if !self.elements.contains_key(element1) {
+    pub fn insert_rpc(&mut self, target: &str, rpc: Rpc) {
```

Comment:
What is the purpose of this function?

Reply:
It is basically a manually RPC generation function. Instead of having nodes with generation rate, which creates too much noise, we place an RPC directly in the receive queue of a particular node. Makes it much easier to test. You can see an example of that in the bookinfo_tests.

Reply:
Gotcha